kbcstorage/base.py

```diff
@@ -0,0 +1,139 @@
+"""
+Base classes for constructing the client.
+
+Primarily exposes a base Endpoint class which deduplicates functionality across
+various endpoints, such as tables, workspaces, jobs, etc. as described in the
+`Storage API documentation`.
+
+
+.. _Storage API documentation:
+    http://docs.keboola.apiary.io/
+"""
+from urllib.parse import urljoin
+
+import requests
+
+
+class Endpoint:
+    """
+    Base class for implementing a single endpoint related to a single entities
+    as described in the Storage API.
+
+    Attributes:
+        path (str): URL for this endpoint.
+        token (str): A key for the Storage API.
+    """
+    def __init__(self, root, extension, token):
+        """
+        Create an endpoint.
+
+        Args
+            root (str): Root url of API. eg.
+                "https://connection.keboola.com/v2/storage/"
+            extension (str): Extension of url for the endpoint. eg. "buckets"
+            token (str): A key for the Storage API. Can be found in the storage
+            console.
+        """
+        self.path = urljoin(root, extension)
+        self.token = token
+
+    def _get(self, params=[], extra_headers={}):
```

Comment:
I think then that the base `Endpoint` `get` and `post` methods don't add any value. I do still think that the base endpoint class is still useful for a few things and I have half implemented some ideas, so hopefully I'll be able to share them in a few days.